tool/tctl/common/plugins_command_test.go

```diff
@@ -0,0 +1,383 @@
+							Credentials: &types.PluginStaticCredentialsSpecV1_APIToken{
+								APIToken: "api-token-goes-here",
+							},
+						},
+					},
+					{
+						ResourceHeader: types.ResourceHeader{
+							Metadata: types.Metadata{
+								Name: "okta-scim-test-scim-token",
+								Labels: map[string]string{
+									types.OktaCredPurposeLabel: types.OktaCredPurposeSCIMToken,
+								},
+							},
+						},
+						Spec: &types.PluginStaticCredentialsSpecV1{
+							Credentials: &types.PluginStaticCredentialsSpecV1_APIToken{
+								APIToken: "scim-token-goes-here",
+							},
+						},
+					},
+				},
+				CredentialLabels: map[string]string{
+					types.OktaOrgURLLabel: "https://example.okta.com",
+				},
+			},
+			expectError: require.NoError,
+		},
+	}
+
+	cmpOptions := []cmp.Option{
+		// Ignore extraneous fields for protobuf bookkeeping
+		cmpopts.IgnoreUnexported(pluginsv1.CreatePluginRequest{}),
+
+		// Ignore any SCIM-token credentials because the bcrypt hash of the token
+		// will change on every run.
+		// TODO: Find a way to only exclude the token hash from the comparison,
+		//       rather than the whole credential
+		cmpopts.IgnoreSliceElements(func(c *types.PluginStaticCredentialsV1) bool {
+			l, _ := c.GetLabel(types.OktaCredPurposeLabel)
+			return l == types.OktaCredPurposeSCIMToken
+		}),
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			var args installPluginArgs
+
+			if testCase.expectRequest != nil {
+				pluginsClient := &mockPluginsClient{}
+				t.Cleanup(func() { pluginsClient.AssertExpectations(t) })
+
+				pluginsClient.
+					On("CreatePlugin", anyContext, mock.Anything, mock.Anything).
+					Run(func(args mock.Arguments) {
+						require.IsType(t, (*pluginsv1.CreatePluginRequest)(nil), args.Get(1))
+						request := args.Get(1).(*pluginsv1.CreatePluginRequest)
+						require.Empty(t, cmp.Diff(testCase.expectRequest, request, cmpOptions...))
+					}).
+					Return(&emptypb.Empty{}, nil)
+
+				args.plugins = pluginsClient
+			}
+
+			if testCase.expectSAMLConnectorQuery != "" {
+				samlConnectorsClient := &mockSAMLConnectorsClient{}
+				t.Cleanup(func() { samlConnectorsClient.AssertExpectations(t) })
+
+				samlConnectorsClient.
+					On("GetSAMLConnector", anyContext, testCase.expectSAMLConnectorQuery, false).
+					Return(&types.SAMLConnectorV2{}, nil)
+
+				args.samlConnectors = samlConnectorsClient
+			}
+
+			ctx, cancel := context.WithCancel(context.Background())
+			t.Cleanup(cancel)
+
+			err := testCase.cmd.InstallOkta(ctx, args)
+			testCase.expectError(t, err)
+		})
+	}
+}
+
+func requireBadParameter(t require.TestingT, err error, _ ...interface{}) {
+	require.Error(t, err)
+	require.True(t, trace.IsBadParameter(err), "Expecting bad parameter, got %T: \"%v\"", err, err)
+}
+
+// must will apply the Go "must" idiom to any arbitrary function that returns a
+// conventional (value, error) pair
+func must[T any](value T, err error) T {
```

Comment:
Since this function is only used to make sure that URL is correct, I'd called it mustURL for example and define as a function not generic.